### PR TITLE
Convert to circle

### PR DESCRIPTION
**Pull request by LinuxBozo**
Conversion from Travis to Circle.

Makes use of https://github.com/tpoechtrager/osxcross to do cross compilation for OSX since Mac support on CircleCI is spotty on v1, and not supported on v2 configurations.

### CODE REVIEW

File: .circleci/build-osx-cross.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (cmc333333) [inline .circleci/build-osx-cross.sh:1]: Do we want a `set -e`/similar here?

File: .circleci/build-osx-cross.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+cd /tmp
+git clone https://github.com/tpoechtrager/osxcross.git
+cd osxcross
+curl -#L https://github.com/phracker/MacOSX-SDKs/releases/download/10.13/MacOSX10.11.sdk.tar.xz -o tarballs/MacOSX10.11.sdk.tar.xz
```
Comment (cmc333333) [inline .circleci/build-osx-cross.sh]: The `-#L` confuses me -- do we not want redirects?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    working_directory: /go/src/github.com/18F/git-seekret
+
+    steps:
+      - checkout
+      - run: sudo sh -c 'echo "deb http://ftp.us.debian.org/debian/ unstable main contrib non-free" >> /etc/apt/sources.list'
+      - run: sudo apt-get update -qq
+      - run: sudo apt-get install -y pkg-config cmake shellcheck clang llvm-dev libxml2-dev uuid-dev libssl-dev file bash patch make gcc g++ zlib1g-dev libmpc-dev libmpfr-dev libgmp-dev tar xz-utils bzip2 gzip sed cpio
+      - run: shellcheck build.sh
+      - run: ./.circleci/build-osx-cross.sh
+      - run: export CROSS_DIR=/tmp/osxcross/target && export PATH=$CROSS_DIR/bin:$PATH && export SDKROOT="$CROSS_DIR/SDK/MacOSX10.11.sdk" && export OSTYPE=darwin15 && ./build.sh
```
Comment (cmc333333) [inline .circleci/config.yml:40]: Small detail, but would this be cleaner (and work) as either
```
CROSS_DIR=/tmp/osxcross/target PATH=$CROSS_DIR/bin:$PATH SDKROOT="$CROSS_DIR/SDK/MacOSX10.11.sdk" OSTYPE=darwin15 ./build.sh
```
(i.e. no exports)

or
```yaml
run:
    environment:
        CROSS_DIR: /tmp/osxcross/target
        PATH: "$CROSS_DIR/bin:$PATH" 
        SDKROOT: "$CROSS_DIR/SDK/MacOSX10.11.sdk" 
        OSTYPE: darwin15
    command: ./build.sh
```
?
The latter is easier to read, imho.

**Review by cmc333333 — APPROVED** [review]
Neat! I skimmed over the code and it fits a shape I'd expect (though I'm not familiar with the domain). I've ran the compiled artifacts for both OSX and Linux and they seemed to execute just fine.

File: .circleci/build-osx-cross.sh (hunk 1 above)
Comment (LinuxBozo) [reply .circleci/build-osx-cross.sh:1]: Good call.

File: .circleci/config.yml (hunk 3 above)
Comment (LinuxBozo) [reply .circleci/config.yml:40]: Good call, I like the latter as well.

File: .circleci/build-osx-cross.sh (hunk 2 above)
Comment (LinuxBozo) [reply .circleci/build-osx-cross.sh]: `-#L` is two options to curl.. progress bar, and follow redirects.  It doesn't actually comment as seen in the syntax highlighting.

Comment (LinuxBozo) [reply .circleci/build-osx-cross.sh]: Will use the long form option instead to clear any confusion in the future.

File: .circleci/config.yml (hunk 3 above)
Comment (LinuxBozo) [reply .circleci/config.yml:40]: And unfortunately, doesn't work for interpolation like we're doing for vars reliant on CROSS_DIR: https://circleci.com/docs/2.0/env-vars/#interpolating-environment-variables-to-set-other-environment-variables

😞

**Pull request merged**